python/lsst/summit/extras/fastStarTrackerAnalysis.py

```diff
@@ -0,0 +1,625 @@
+            )
+        return False
+
+    consistent = True
+    toPrint = []
+    nSources = set([sourceSet[0].nSourcesInImage for sourceSet in results])
+    if len(nSources) != 1:
+        toPrint.append(f"❌ Images contain a variable number of sources: {nSources}")
+        consistent = False
+    else:
+        n = nSources.pop()
+        toPrint.append(f"✅ All images contain the same nominal number of sources at detection stage: {n}")
+
+    nSourcesCounted = set([len(sourceSet) for sourceSet in results])
+    if len(nSourcesCounted) != 1:
+        toPrint.append(
+            f"❌ Number of actual sources in each sourceSet varies, got: {nSourcesCounted}."
+            " If some were manually removed you can ignore this"
+        )
+        consistent = False
+    else:
+        n = nSourcesCounted.pop()
+        toPrint.append(f"✅ All results contain the same number of actual sources per image: {n}")
+
+    widths = set([sourceSet[0].parentImageWidth for sourceSet in results])
+    heights = set([sourceSet[0].parentImageHeight for sourceSet in results])
+    if len(widths) != 1 or len(heights) != 1:
+        toPrint.append(f"❌ Input images were of variable dimenions! {widths=}, {heights=}")
+        consistent = False
+    else:
+        toPrint.append("✅ All input images were of the same dimensions")
+
+    if len(results) > 1:  # can't np.diff an array of length 1 so these are not useful/defined
+        # now the basic checks have passed, do some sanity checks on the
+        # maximum deltas for the primary sources
+        sources = [sourceSet[0] for sourceSet in results]
+        dx = np.diff([s.centroidX for s in sources])
+        dy = np.diff([s.centroidY for s in sources])
+        maxMovementX = np.max(dx)
+        maxMovementY = np.max(dy)
+        happyOrSad = "✅"
+        if max(maxMovementX, maxMovementY) > maxAllowableShift:
+            consistent = False
+            happyOrSad = "❌"
+
+        toPrint.append(
+            f"{happyOrSad} Maximum centroid movement of brightest object between images in (x, y)"
+            f" = ({maxMovementX:.2f}, {maxMovementY:.2f}) pix"
+        )
+
+        fluxStd = np.nanstd([s.rawFlux for s in sources])
+        fluxMean = np.nanmean([s.rawFlux for s in sources])
+        toPrint.append(f"Mean and stddev of flux from brightest object = {fluxMean:.1f} ± {fluxStd:.1f} ADU")
+
+    if not silent:
+        for line in toPrint:
+            print(line)
+
+    return consistent
+
+
+def plotSourceMovement(results, sourceIndex=0, allowInconsistent=False):
+    """Plot the centroid movements and fluxes etc for a set of results.
+
+    By default the brightest source in each image is plotted, but this can be
+    changed by setting ``sourceIndex`` to values greater than 0 to move through
+    the list of sources in each image.
+
+    Parameters
+    ----------
+    results : `dict` of `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```

Comment:
`lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` -> `lsst.summit.extras.fastStarTrackerAnalysis.Source` ?

Reply:
Yup. Done here and everywhere.